### PR TITLE
Cockpit 216

### PR DESCRIPTION
**Pull request by allisonkarlitskaya**
Preview: https://allisonkarlitskaya.github.io/cockpit-project.github.io/blog/cockpit-216.html

### CODE REVIEW

**Comment by marusak** [comment]
The screenshot for selinux is a bit out of date, the formating of solutions also has been updated.
Let me send you a new one.

**Comment by marusak** [comment]
newslinux
The new screenshot.

> The SELinux page now has “Apply this solution” buttons for suggested solutions to logged access control errors.

Not completely true, it had it even before, the change is that now it is possible to apply the solution with click of a button in more cases.

> The System Modifications table will be updated after applying a solution.

That is correct and is a new feature, but not sure if we need to propagate it, as it is something, which is expected to happen :)

Additional change is enhanced displaying of solutions for better readability and easier copying of solution commands.

**Comment by allisonkarlitskaya** [comment]
Would it be possible to get a screenshot a bit less wide?

**Comment by marusak** [comment]
abitlesswide
After that it starts to wrap. Is this better?

File: _posts/2020-04-01-cockpit-216.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+summary: Cockpit with improved SELinux page and last login banner
+comments: 'true'
+---
+
+Cockpit is the [modern Linux admin interface](https://cockpit-project.org/). We release regularly. Here are the release notes from version 216.
+
+### SELinux: Automatic application of solutions
+
+The SELinux page now has "Apply this solution" buttons for suggested
+solutions to logged access control errors.
```
Comment (martinpitt) [inline _posts/2020-04-01-cockpit-216.md:17]: Perhaps "The SELinux page can now automatically apply solutions that set [SELinux Booleans](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/selinux_users_and_administrators_guide/sect-security-enhanced_linux-working_with_selinux-booleans)." ?

File: _posts/2020-04-01-cockpit-216.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+![Apply this solution](/images/selinux-apply-this-solution.png)
+
+The System Modifications table will be updated after applying a
```
Comment (martinpitt) [inline _posts/2020-04-01-cockpit-216.md:21]: Maybe add before this:  "In addition, the command lines in solution details are now formatted properly."

File: _posts/2020-04-01-cockpit-216.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+This release drops the virsh backend support from the machines page and
+switches over to using libvirt-dbus exclusively.  As a result, as of this
+release, cockpit-machines is no longer supported on Ubuntu 18.04.
+
+### New "last login" banner
+
+Cockpit now displays a banner on login similar to the banner displayed
+when logging in via text console or SSH which shows the last successful
+login time, plus information about any failed login attempts.
```
Comment (martinpitt) [inline _posts/2020-04-01-cockpit-216.md:34]: This is a rather long sentence, I think it should be split up.

File: _posts/2020-04-01-cockpit-216.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+The System Modifications table will be updated after applying a
+solution.
+
+### Machines: Drop virsh backend support
+
+This release drops the virsh backend support from the machines page and
+switches over to using libvirt-dbus exclusively.  As a result, as of this
+release, cockpit-machines is no longer supported on Ubuntu 18.04.
+
+### New "last login" banner
```
Comment (martinpitt) [inline _posts/2020-04-01-cockpit-216.md:30]: I'd prefix with "Overview:" to keep the "Page: What" pattern. I'd  also drop "New", as everything in release notes is supposed to be new. :-)

File: _posts/2020-04-01-cockpit-216.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+Cockpit is the [modern Linux admin interface](https://cockpit-project.org/). We release regularly. Here are the release notes from version 216.
+
+### SELinux: Automatic application of solutions that set booleans
+
+The SELinux page can now automatically apply solutions that set [SELinux booleans](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/selinux_users_and_administrators_guide/sect-security-enhanced_linux-working_with_selinux-booleans).
+
+![Apply this solution](/images/selinux-apply-this-solution.png)
+
+In addition, the command lines in solution details are now formatted properly.
```
Comment (garrett) [inline _posts/2020-04-01-cockpit-216.md:20]:
```suggestion
In addition, command lines in solution details are now formatted properly.
```

File: _posts/2020-04-01-cockpit-216.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+### Machines: Drop virsh backend support
+
+This release drops the virsh backend support from the machines page and
+switches over to using libvirt-dbus exclusively.  As a result, as of this
+release, cockpit-machines is no longer supported on Ubuntu 18.04.
```
Comment (garrett) [inline _posts/2020-04-01-cockpit-216.md:26]:
```suggestion
This release drops virsh backend support from the Machines page and
switches to using libvirt-dbus exclusively.  As a result, as of this
release, cockpit-machines is no longer supported on Ubuntu 18.04.
```

File: _posts/2020-04-01-cockpit-216.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+record of failed login attempts).
+
+### Try it out
+
+Cockpit 216 is available now:
+
+ * [For your Linux system](https://cockpit-project.org/running.html)
+ * [Source Tarball](https://github.com/cockpit-project/cockpit/releases/tag/216)
+ * [Fedora 31](https://bodhi.fedoraproject.org/updates/FEDORA-2020-TODO)
+ * [Fedora 32](https://bodhi.fedoraproject.org/updates/FEDORA-2020-TODO)
```
Comment (garrett) [inline _posts/2020-04-01-cockpit-216.md:48]:
```suggestion
 * [Fedora 32](https://bodhi.fedoraproject.org/updates/FEDORA-2020-TODO)
 
 *[SELinux]: security-enhanced Linux, a way to ensure better access control to resources
 *[virsh]: virtual machine shell, a utility to control virtual machines at the command line
 *[libvirt-dbus]: a way to control virtual machines over the common "d-bus" inter-program communication layer 
*[SSH]: secure shell
```

Added some definnitions. Trying to explain an API like d-bus in simpler English is difficult. I might've gotten it wrong. Please fix. :wink:

**Comment by garrett** [comment]
Also, a screenshot convention is to have a border around screenshots that are partial (with cropping) or a resized screenshot. When we show a complete block excerpt, we usually don't have an outline.

(FWIW: There is a class to apply an outline, but we do sometimes bake them into the image themselves too. It's not really important either way.)

File: _posts/2020-04-01-cockpit-216.md (hunk 7 above)
Comment (martinpitt) [reply _posts/2020-04-01-cockpit-216.md:48]: It's "D-Bus", and "inter-process" communication, but otherwise looks  good, thanks!

**Comment by garrett** [comment]
I've adjusted the crop of the images (adding more spacing to the top of each), lightened the border on the first image, and optimized both (for file size savings).

**Comment by martinpitt** [comment]
bodhi URLs are available now, release finished: https://bodhi.fedoraproject.org/updates/?packages=cockpit

**Pull request merged**